snake/snake_game_khr.py

```diff
@@ -0,0 +1,198 @@
+            (0, -1)
+        ]
+        result = []
+        for next_position in next_positions:
+            new_positions = self.body[-1][0] + next_position[0], self.body[-1][1] + next_position[1]
+            for new_position in new_positions:
+                if new_position in self.body:
+                    continue
+            result.append(new_positions)
+        # print(result)
+        # rand_result = result[random.randrange(len(result))]
+        # return choices that do not intersect with snake body coords
+        return result
+
+
+class Apple:
+    def __init__(self, symbol='$', position=(2, 2)):
+        self.position = position
+        self.symbol = symbol
+
+
+class GameOverError(Exception):
+    pass
+
+
+class Game:
+    def __init__(self, width, height):
+        self.symbol = "o"
+        self.width = width
+        self.height = height
+        self.board = Board(self.width, self.height)
+        self.snake = Snake()
+        self.apple = Apple()
+
+    def init_apple(self):
+        last_pos = self.apple.position
+        self.board = Board(self.width, self.height)
+        positions = []
+        for x in range(self.height):
+            for y in range(self.width):
+                z = (item for item in self.snake.body)
+                if (x, y) != last_pos and (x, y) not in z:
+                    if self.board.board[x][y] == ' ':
+                        positions.append((x, y))
+
+        if positions:
+            position = random.choices(positions)[0]
+            self.apple = Apple(position=position, symbol="$")
+            return self.apple.position
+        else:
+            raise GameOverError('No places for apple!')
+
+    def play(self):
+        game_cycles = 20
+        try:
+            self.render()  # initial render of board, snake and apple
+            while game_cycles > 0:  # start our game
+                snake_i, snake_j = self.snake.body[-1]  # remember last position of snake's head
+                move = None
+                # TODO: find next possible position (next_move) for snake to move
+                possible_moves = self.snake.choices()
+                next_moves = []
+                for move in possible_moves:
+                    # print(next_move, type(next_move))
+                    if move[0] != 0 and move[1] != 0:
+                        next_moves.append(move)
+                # print(next_moves)
+                distances = []
+                for move in next_moves:
+                    dist = self.board.distance(move, self.apple.position)
+                    distances.append(dist)
+                # print(distances)
+                res = {}
+                for key in next_moves:
+                    for value in distances:
+                        res[key] = value
+                        distances.remove(value)
+                        break
+                # print(str(res))
+                next_move = min(res, key=res.get)
+                # print(next_move)
```

Comment:
`while` below and a lot of movements does not make a lot of sense to me, maybe this will fix it a bit:
```py
                print(str(res))
                next_move = min(res, key=res.get)
                
                if next_move == self.apple.position:
                    # TODO: eat apple, render this action, init new apple and render it on board
                    # self.snake.body.append(next_move)
                    self.snake.eat(next_move)
                    self.render()
                    self.init_apple()
                    self.render()
                    continue
                self.snake.move(next_move)
                self.render()
                if next_move in self.snake.body:
                    GameOverError('Game over:(')
                if next_move is None:  # there is no possible move for snake
                    self.snake.move((snake_i + 1, snake_j + 1))  # just do one move forward for snake
                    self.render()  # render last move
                    GameOverError('No moves for snake!')  # end the game (snake does not have next valid move)
                game_cycles -= 1
```